data/ignition/files/usr/local/bin/start-agent.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+source common.sh
+
+echo "Waiting for infra-env-id to be available"
+INFRA_ENV_ID=""
+until [[ $INFRA_ENV_ID != "" && $INFRA_ENV_ID != "null" ]]; do
+    sleep 5
+    printf 'Querying assisted-service for infra-env-id...'
```

Comment:
Any reason for using printf here over echo?
In any case, you might want to make it go to stderr
```suggestion
    printf 'Querying assisted-service for infra-env-id...'
```

Reply:
Should probably have a `\n` if using printf

Reply:
In a previous iteration the printf was being used to format a result. It should be changed to echo now.